sonarts-core/src/rules/noSelfAssignmentRule.ts

```diff
@@ -75,22 +76,21 @@ class Walker extends tslint.ProgramAwareRuleWalker {
       this.isCallExpression(right) &&
       this.isPropertyAccessExpression(right.expression) &&
       this.isIdentifier(right.expression.expression) &&
-      this.isArray(right.expression.expression) &&
-      right.expression.name.text === "reverse" &&
+      this.isArrayMutatingCall(right.expression) &&
       right.expression.expression.text === left.text
     );
   }
 
+  private isArrayMutatingCall(expression: ts.PropertyAccessExpression): boolean {
```

Comment:
@michalb-sonar how about making this method a public function and reuse it from another check?..